### PR TITLE
Rewrite some non-failing xNodeBuilder tests so they dont print mismatch messages

### PR DESCRIPTION
**Pull request by sepidehMS**
Some non-failing xNodeBuilder tests printed irrelevant error messages. I rewrote the tests so that they don't use XmlDiff.Compare and compare the results directly.

### CODE REVIEW

File: src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs [hunk 1]
```diff
@@ -3599,17 +3603,21 @@ public void comment_5()
...
                     {
-                        if (CompareReader(doc, "<Root><!--test - - --></Root>"))
+                        string expectedXml = "<Root><!--test - - --></Root>";
+
+                        using (XmlReader reader = doc.CreateReader())
                         {
-                            throw new TestException(TestResult.Failed, "");
-                        } //by design
+                            Assert.Equal(expectedXml, MoveToFirstElement(reader).ReadOuterXml());
+                        }
```
Comment (stephentoub) [inline src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs]: What's the (pre-existing) ```catch (ArgumentException) { return; }``` for?  Seems like that could eat a bunch of errors silently.

**Comment by stephentoub** [comment]
LGTM

File: src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs (hunk 1 above)
Comment (sepidehMS) [inline src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs]: It's not clear to me why this has been added. But currently, this line is not hit in any of these tests.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by sepidehMS** [comment]
@dotnet-bot test this please

**Comment by sepidehMS** [comment]
@dotnet-bot test this please

**Pull request merged**